test/test_memory_layout.cc

```diff
@@ -111,7 +111,7 @@ BOOST_AUTO_TEST_CASE(test_layout_arrays)
         Arrays arrays_v;
         size_t expected[] = {
             MemLayout::FLAG_MEMCPY,
-               reinterpret_cast<uint8_t*>(&arrays_v.a_v_numeric) - reinterpret_cast<uint8_t*>(&arrays_v),
+               static_cast<ptrdiff_t>(reinterpret_cast<uint8_t*>(&arrays_v.a_v_numeric) - reinterpret_cast<uint8_t*>(&arrays_v)),
```

Comment:
Not what I meant.... Sorry :P AFAIK, taking the difference of two pointers in C++ is supposed to return std::ptrdiff_t, so by changing the type of expected[] you should be able to remove the static_cast completely. No ?